client/state/signup/steps/website-content/reducer.ts

```diff
@@ -44,6 +45,41 @@ export default withSchemaValidation(
 					...state,
 					currentIndex: action.payload,
 				};
+			case SIGNUP_STEPS_WEBSITE_CONTENT_LOGO_UPLOAD_FAILED:
+			case SIGNUP_STEPS_WEBSITE_CONTENT_LOGO_UPLOAD_STARTED: {
```

Comment:
Would it be easier to split this reducer into separate reducers and use `combineReducers`?

For example:
```
export const pages = withSchemaValidation(<... pages reducer ...>);
export const currentIndex = withSchemaValidation(<... currentIndex reducer ...>);
export const imageUploadStates = <... imageUploadStates reducer ...>; // We don't need to persist this state
export const siteLogoUrl = withSchemaValidation(<... siteLogoUrl reducer ...>);

export default combineReducers({
  pages,
  currentIndex,
  imageUploadStates,
  siteLogoUrl,
})
```

What do you think?

Reply:
You do have a point 💯  the state tree seems to be growing unsustainably. This will require some effort though since the shape will change and it will be effecting a lot of areas. I can pick this up right after this change in a seperate focussed PR🥇